src/ts/storageutils.ts

```diff
@@ -39,8 +17,13 @@ export namespace StorageUtils {
    * @param data the item's data
    */
   export function setItem(key: string, data: string): void {
-    if (StorageUtils.hasLocalStorage()) {
-      window.localStorage.setItem(key, data);
+    try {
+      if (StorageUtils.isLocalStorageAvailable()) {
+        window.localStorage.setItem(key, data);
+      }
```

Comment:
the `isLocalStorageAvailable()` check does not require try-catch, that's already done internally. So we could pull it out of the try block and do an early return, or wrap the whole try-catch block with it. (same for `getItem()`)

Reply:
Ah yes, you are right. I thought that, when browser prohibits access, exceptions are triggered when actually trying to call `getItem` / `setItem`, but it already happens when trying to access `window.localStorage`. Hence, it's already covered in `isLocalStorageAvailable`.

I removed the `try-catch`es in this commit: [Removed superfluous try-catch](https://github.com/bitmovin/bitmovin-player-ui/pull/612/commits/3c2df0690034ff24b315e7a3820e13e4970f0571)

Let's have a call if you meant it in a different way :)